### PR TITLE
[ROCm] fixed rocm kernel link

### PR DESCRIPTION
**Pull request by i-chaochen**
Fixed a recent build error due to the missing link at rocm gpu executor and some updates on rocm_driver

@akuegel @ddunl  Thanks in advance!

### CODE REVIEW

**Comment by ddunl** [comment]
Do you know which change caused the error out of curiosity? I know there's been a lot of changes to stream_executor recently and wondering if those are causing issues for you

**Review by ddunl — APPROVED** [review]

**Comment by i-chaochen** [comment]
> Do you know which change caused the error out of curiosity? I know there's been a lot of changes to stream_executor recently and wondering if those are causing issues for you

I think it's something related to https://github.com/openxla/xla/commit/d04387fd4075af9d70858b2910cfa19a28a28fee and this is a follow-up based on https://github.com/openxla/xla/commit/5c405c4a25d7f32036c03995383462d6ce9a5619

it seems rocm gpu executor now requires `//xla/stream_executor:kernel` but cuda doesn't need it (or maybe it has it in somewhere else)
 
https://github.com/openxla/xla/pull/5867/files#diff-923ec2612240244a60ae28c17f798a7daeccbab84a8e9de9efd7ba96c311c763R107

**Comment by ddunl** [comment]
Ok gotcha, makes sense, unfortunately I'd say that there's still probably some churn left to go for the rocm code then as I know of at least 1 more change that will be similar-ish (and therefore may have similar unintended consequences for rocm code). Sorry, and thanks for the PR!!

**Comment by i-chaochen** [comment]
Thanks for the notice. Yes we are monitoring it by our daily nightly CI and we will try our best to catch up.

**Comment by ddunl** [comment]
I think it's hard to predict which changes will affect rocm honestly just because the changes are untangling the messy stream_executor setup, so I'm not sure if the most recent change will have negative effects, but I wouldn't be surprised either. I think I'll try to merge this as is and then fingers crossed there won't be another breakage (but we can merge another small one if needed of course)

File: xla/stream_executor/rocm/BUILD [hunk 1]
```diff
@@ -306,7 +307,7 @@ cc_library(
         "//xla/stream_executor:event",
         "//xla/stream_executor:plugin_registry",
         "//xla/stream_executor:scratch_allocator",
-        "//xla/stream_executor:stream_executor_pimpl_header",
```
Comment (ezhulenev) [inline xla/stream_executor/rocm/BUILD:309]: This likely has to be `//xla/stream_executor` or `//xla/stream_executor:stream_executor_impl`, I'll be removing `pimpl` targets shortly

File: xla/stream_executor/rocm/BUILD [hunk 2]
```diff
@@ -104,6 +104,7 @@ cc_library(
         ":rocm_platform_id",
         "@com_google_absl//absl/functional:any_invocable",
         "@com_google_absl//absl/strings",
+        "//xla/stream_executor:kernel",
```
Comment (ezhulenev) [inline xla/stream_executor/rocm/BUILD:107]: `kernel` will be an internal target with private visibility, it should be just `//xla/stream_executor` to bring all SE dependencies

**Comment by ezhulenev** [comment]
I'm working on cleaning up StreamExecutor targets and headers, the end state is:

1. `//xla/stream_executor` - this is StreamExecutor implementation that most (all) of the users should depend on. I don't think there is a reason for any ROCM target not to depend on it.
2. `//xla/stream_executor:stream_executor_headers` - only headers for public StreamExecutor APIs, this target will be visible for all subpackages, and a few external clients that need it (Tensorflow libraries)
3. `device_options`, `platform` and few other small libraries that do not depend on StreamExecutor will be public

All other targets (`kernel`, `event`, etc.) will be private to stream_executor package, and considered implementation detail.

`pimpl` headers and targets will be deleted.

**Comment by i-chaochen** [comment]
Thanks @ezhulenev I will recheck it tomorrow when you're finished.

**Pull request closed without merging**